airflow/providers/databricks/plugins/databricks_workflow.py

```diff
@@ -0,0 +1,472 @@
+
+        :return: A list of task IDs for tasks that have failed or have been skipped.
+        """
+        return [
+            t.task_id
+            for t in dr.get_task_instances(
+                state=[
+                    TaskInstanceState.FAILED,
+                    TaskInstanceState.SKIPPED,
+                    TaskInstanceState.UP_FOR_RETRY,
+                    TaskInstanceState.UPSTREAM_FAILED,
+                    None,
+                ],
+            )
+        ]
+
+
+class WorkflowJobRepairSingleTaskLink(BaseOperatorLink, LoggingMixin):
+    """Construct a link to send a repair request for a single databricks task."""
+
+    name = "Repair a single task"
+
+    def get_link(
+        self,
+        operator,
+        dttm=None,
+        *,
+        ti_key: TaskInstanceKey | None = None,
+    ) -> str:
+        if not ti_key:
+            ti = get_task_instance(operator, dttm)
+            ti_key = ti.key
+
+        task_group = operator.task_group
+        if not task_group:
+            raise AirflowException("Task group is required for generating repair link.")
+
+        self.log.info(
+            "Creating link to repair a single task for databricks job run %s task %s",
+            task_group.group_id,
+            ti_key.task_id,
+        )
+        dag = airflow_app.dag_bag.get_dag(ti_key.dag_id)
+        task = dag.get_task(ti_key.task_id)
+
+        if ".launch" not in ti_key.task_id:
+            launch_task_id = get_launch_task_id(task_group)
+            ti_key = _get_launch_task_key(ti_key, task_id=launch_task_id)
+        metadata = get_xcom_result(ti_key, "return_value")
+
+        query_params = {
+            "dag_id": ti_key.dag_id,
+            "databricks_conn_id": metadata.conn_id,
+            "databricks_run_id": metadata.run_id,
+            "run_id": ti_key.run_id,
+            "tasks_to_repair": _get_databricks_task_id(task),
+        }
+        return url_for("RepairDatabricksTasks.repair", **query_params)
+
+
+class RepairDatabricksTasks(AirflowBaseView, LoggingMixin):
+    """Repair databricks tasks from Airflow."""
+
+    default_view = "repair"
+
+    @expose("/repair_databricks_job/<string:dag_id>/<string:run_id>/", methods=("GET",))
+    @auth.has_access_dag("POST", DagAccessEntity.RUN)
+    def repair(self, dag_id: str, run_id: str):
+        if not dag_id:
+            return_url = request.referrer or url_for("Airflow.index")
+            flash("No dag_id provided. Cannot repair tasks.")
+            return redirect(return_url)
+
+        view = conf.get("webserver", "dag_default_view")
+        return_url = self._get_return_url(dag_id, view)
+
+        tasks_to_repair = request.values.get("tasks_to_repair")
+        self.log.info("Tasks to repair: %s", tasks_to_repair)
+        if not tasks_to_repair:
+            flash("No tasks to repair. Not sending repair request.")
+            return redirect(return_url)
```

Comment:
Why are you not adding the tasks to repair to the URL pattern above? It is required anyway.

Reply:
I tried initially with adding dag_id and run_id to the URL route. Besides tasks_to_repair needs to be a list of values and not sure if it could fit it in the URL route

Reply:
I have added dag_id and run_id, but not fully convinced with adding tasks_to_repair in the url route. 

Flask does not have built-in url map converter for list type values in the url route. If we have to have it in the URL, we will have to add a custom converter https://flask.palletsprojects.com/en/3.0.x/api/#flask.Flask.url_map and adding it to the app from a plugin seems risky and could cause potential side-effects. I guess having a validation feels safer, IMO. WDYT?

The default url_map converters with Flask
<img width="1720" alt="Screenshot 2024-07-12 at 9 31 47 PM" src="https://github.com/user-attachments/assets/6edad489-ff6f-46e5-a6d6-3f65c5c65c91">